mapper/orm/src/main/java/org/hibernate/search/mapper/orm/model/impl/HibernateOrmPropertyModel.java

```diff
@@ -93,8 +93,7 @@ public String getName() {
 	public PropertyHandle<T> getHandle() {
 		if ( handle == null ) {
 			try {
-				// TODO HSEARCH-3058
-				handle = (PropertyHandle<T>) introspector.createPropertyHandle( name, member );
+				handle = introspector.createPropertyHandle( name, member );
```

Comment:
This effectively hides the problem under the carpet. The problem is that the introspector (and, below that, the factory) cannot guarantee that the property handle will be of type `PropertyHandle<T>` for any member. With this solution you could pass a method representing `toString` and request a `PropertyHandle<Integer>` and that would compile just fine. At runtime you would get an exception eventually, but it would compile.

There's no nice solution here, but I think there's a more readable one: remove this entire commit, and just replace the beginning of this method with this:

```
@SuppressWarnings("unchecked") // By construction, we know the member returns values of type T
public PropertyHandle<T> getHandle() {
		if ( handle == null ) {
			try {
				handle = (PropertyHandle<T>) introspector.createPropertyHandle( name, member );
```

And do something similar in `org.hibernate.search.mapper.javabean.model.impl.JavaBeanPropertyModel#getHandle`.